azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/DeploymentApiLiveTest.java

```diff
@@ -0,0 +1,127 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.features;
+
+import org.jclouds.azurecompute.arm.domain.Deployment;
+import org.jclouds.azurecompute.arm.domain.Deployment.ProvisioningState;
+import org.jclouds.azurecompute.arm.internal.BaseAzureComputeApiLiveTest;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.List;
+
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertNotNull;
+
+@Test(groups = "live", testName = "DeploymentApiLiveTest", singleThreaded = true)
+public class DeploymentApiLiveTest extends BaseAzureComputeApiLiveTest {
+    private int maxTestDuration = 90;
+    private int pollingInterval = 3; // how frequently to poll for create status
+
+    private String resourceGroup;
+    private String resourceName;
+    private String deploymentName;
+
+    @BeforeClass
+    @Override
+    public void setup() {
+        super.setup();
+        resourceGroup = getResourceGroupName();
+    }
+
+    private String getPutBody(String template, String mode, String parameters) {
+        String body = "{ " +
+                      "\"properties\" : " +
+                      "  { " +
+                      "    \"template\" : " + template + ", " +
+                      "    \"mode\" : \"" + mode + "\", " +
+                      "    \"parameters\" : " + parameters + " " +
+                      "  } " +
+                      "}";
+        return body;
+    }
+
+    @Test(groups = "live")
+    public void testCreate() {
+        Long now = System.currentTimeMillis();
+        resourceName = "jcres" + now;
+        deploymentName = "jcdep" + now;
+        String template = "{\"$schema\":\"https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#\",\"contentVersion\":\"1.0.0.0\",\"parameters\":{\"newStorageAccountName\":{\"type\":\"string\",\"metadata\":{\"description\":\"Name of the Storage Account\"}},\"storageAccountType\":{\"type\":\"string\",\"defaultValue\":\"Standard_LRS\",\"allowedValues\":[\"Standard_LRS\",\"Standard_GRS\",\"Standard_ZRS\"],\"metadata\":{\"description\":\"Storage Account type\"}},\"location\":{\"type\":\"string\",\"allowedValues\":[\"East US\",\"West US\",\"West Europe\",\"East Asia\",\"Southeast Asia\"],\"metadata\":{\"description\":\"Location of storage account\"}}},\"resources\":[{\"type\":\"Microsoft.Storage/storageAccounts\",\"name\":\"[parameters('newStorageAccountName')]\",\"apiVersion\":\"2015-05-01-preview\",\"location\":\"[parameters('location')]\",\"properties\":{\"accountType\":\"[parameters('storageAccountType')]\"}}]}";
+        String parameters = "{\"newStorageAccountName\":{\"value\":\"" + resourceName + "\"},\"storageAccountType\":{\"value\":\"Standard_LRS\"},\"location\":{\"value\":\"West US\"}}";
+        String properties = getPutBody(template, "Incremental", parameters);
```

Comment:
> But the user experience, in this case is worse: Users have to build on their own that json. They can use the builders, but they have to manually serialize the template into a json, just because our DeploymentApi methods expect the template to be a json String.

I agree the user experience is not great today, but this is how users are interfacing with the Azure ARM template deployment process today and this is the intended behavior of the original Deployment API. From the official API doc, the template "Specifies the JSON definition for the template. You use this element when you want to pass the template syntax directly in the request rather than link to an existing template."

If users can call the DeploymentApi directly, then in this case, users will expect to pass in a raw json template to the api much like how they interface with the original api anywhere whether it's thru a REST api call, or thru the Azure CLI, or thru an SDK. [Here is the reference](https://azure.microsoft.com/en-us/documentation/articles/resource-group-authoring-templates/)

Given that this is the design of this particular Deployment API and the Azure ARM template deployment process, we should probably follow the intent of the original api and ensure we give users the same experience as they expect it elsewhere. 

However if users use this thru the jclouds api, then we want to abstract all this raw template business by providing developers with the `DeploymentTemplateBuilder` so they do not have to deal with creating the raw json but instead pass in the parameters and we will build the json template for them.

@nacx WDYT? The current implementation allows users to use the DeploymentApi directly with the raw json template as intended by the original api, but it also supports abstracting the raw json template with a builder if users use jclouds apis.